operator/v1/types_baremetal.go

```diff
@@ -0,0 +1,68 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Metal3ProvisioningController contains configuration used by the Provisioning service (Ironic)
+// to provision baremetal hosts.
+//
+type Metal3ProvisioningController struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the
+	// Metal3ProvisioningController.
+	Spec Metal3ProvisioningControllerSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the
+	// Metal3ProvisioningController.
+	Status Metal3ProvisioningControllerStatus `json:"status,omitempty"`
+}
+
+// ProvisioningDHCP
+type ProvisioningDHCP struct {
+	// managementState indicates whether the DHCP service would be started within
+	// the metal3 cluster or would use an external DHCP service
+	ManagementState ManagementState `json:"managementState"`
+
+	// DHCPRange
+	DHCPRange string `json:"DHCPRange,omitempty"`
+}
+
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type Metal3ProvisioningControllerSpec struct {
+	// provisioningInterface is the name of the network interface on a Baremetal server connected
+	// to the provisioning network.
+	ProvisioningInterface string `json:"provisioningInterface"`
```

Comment:
@damemi we need to explicitly say required in our repo, right?

Reply:
@deads2k yup, because we set `operator/v1` to default optional at the package level https://github.com/openshift/api/blob/master/operator/v1/doc.go#L5

Reply:
> @damemi we need to explicitly say required in our repo, right?

@damemi  Would adding this "// +kubebuilder:object:root=true" take care of that?